humility-core/src/arch_rv.rs

```diff
@@ -0,0 +1,249 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::arch::{instr_operands, Register, RegisterField};
+use anyhow::Result;
+use num_traits::ToPrimitive;
+use std::collections::BTreeMap;
+use strum_macros::EnumIter;
+
+#[allow(non_camel_case_types)]
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Hash,
+    FromPrimitive,
+    ToPrimitive,
+    PartialEq,
+    Eq,
+    Ord,
+    PartialOrd,
+    EnumIter,
+)]
+
+///
+/// See table 3.3 in the riscv debug spec.
+/// (https://raw.githubusercontent.com/riscv/riscv-debug-spec/master/riscv-debug-stable.pdf#table.3.3)
+/// The pc is read through the DPC register
+///
+/// THOU MUST ORDER THE REGISTERS ACCORDING TO ADDRESS!
+///
+pub enum RVRegister {
+    MSTATUS = 0x300,
+    MTVEC = 0x301,
+    MIE = 0x302,
+    MEPC = 0x341,
+    MCAUSE = 0x342,
+    MIP = 0x344,
+    PMPCFG0 = 0x3a0,
+    PMPCFG1,
+    PMPCFG2,
+    PMPCFG3,
+    PMPADDR0 = 0x3b0,
+    PMPADDR1,
+    PMPADDR2,
+    PMPADDR3,
+    PMPADDR4,
+    PMPADDR5,
+    PMPADDR6,
+    PMPADDR7,
+    PMPADDR8,
+    PMPADDR9,
+    PMPADDR10,
+    PMPADDR11,
+    PMPADDR12,
+    PMPADDR13,
+    PMPADDR14,
+    PMPADDR15,
```

Comment:
If we're going to add _some_ PMP registers, I think we might as well add _all_ of them (i.e. `pmpcfg{0..15}` and `pmpaddr{0..63}`).